src/main/java/com/autotune/analyzer/utils/GenerateRecommendation.java

```diff
@@ -38,68 +40,65 @@
-                                    config.put(AnalyzerConstants.ResourceSetting.requests, requestsMap);
-                                    HashMap<AnalyzerConstants.RecommendationItem, RecommendationConfigItem> limitsMap = new HashMap<>();
-                                    limitsMap.put(AnalyzerConstants.RecommendationItem.cpu, getCPUMaxRecommendation(filteredResultsMap));
-                                    limitsMap.put(AnalyzerConstants.RecommendationItem.memory, getMemoryMaxRecommendation(filteredResultsMap));
-                                    config.put(AnalyzerConstants.ResourceSetting.limits, limitsMap);
-                                    Double hours = filteredResultsMap.values().stream().map((x) -> (x.getDurationInMinutes()))
-                                            .collect(Collectors.toList())
-                                            .stream()
-                                            .mapToDouble(f -> f.doubleValue()).sum() / 60;
-                                    recommendation.setDuration_in_hours(hours);
-                                    recommendation.setConfig(config);
-                                    recommendationPeriodMap.put(recPeriod, recommendation);
-                                } else {
-                                    RecommendationNotification notification = new RecommendationNotification(
-                                            AnalyzerConstants.RecommendationNotificationTypes.INFO.getName(),
-                                            AnalyzerConstants.RecommendationNotificationMsgConstant.NOT_ENOUGH_DATA);
-                                    recommendationPeriodMap.put(recPeriod, new Recommendation(notification));
+                        // TODO: Add all possible cases which are added in RecommendationCategory
+                        switch (recommendationCategory) {
+                            case DURATION_BASED:
+                                for (RecommendationSubCategory recommendationSubCategory : recommendationCategory.getRecommendationSubCategories()) {
+                                    DurationBasedRecommendationSubCategory durationBasedRecommendationSubCategory = (DurationBasedRecommendationSubCategory) recommendationSubCategory;
+                                    String recPeriod = durationBasedRecommendationSubCategory.getSubCategory();
+                                    int days = durationBasedRecommendationSubCategory.getDuration();
+                                    monitorStartDate = addDays(monitorEndDate, -1 * days);
+                                    if (monitorStartDate.compareTo(minDate) >= 0 || days == 1) {
+                                        Timestamp finalMonitorStartDate = monitorStartDate;
+                                        System.out.println(finalMonitorStartDate);
+                                        System.out.println(monitorEndDate);
```

Comment:
This should be a `LOGGER.debug`

Reply:
They look like print statements added to check the flow while dev, I have removed the print statements.